src/uct/tcp/tcp_ep.c

```diff
@@ -898,11 +915,8 @@ static inline void uct_tcp_ep_handle_put_ack(uct_tcp_ep_t *ep,
                                             uct_tcp_iface_t);
     uct_tcp_ep_put_completion_t *put_comp;
 
-    if (put_ack->sn == ep->tx.put_sn) {
-        /* Since there are no other PUT operations in-flight, can remove flag
-         * and decrement iface outstanding operations counter */
-        ucs_assert(ep->flags & UCT_TCP_EP_FLAG_PUT_TX_WAITING_ACK);
-        ep->flags &= ~UCT_TCP_EP_FLAG_PUT_TX_WAITING_ACK;
+    ucs_assert(ep->tx.put_cnt != 0);
+    if (--ep->tx.put_cnt == 0) {
```

Comment:
why needed to change this logic?

Reply:
since if some operation AM operations done after PUT operations are scheduled, `ep->tx.sn` != `put_ack->sn`